udf_js/json_extract_missing_cols.sql

```diff
@@ -37,43 +37,59 @@ LANGUAGE js AS """
         var next_val = next_keypath.reduce((obj, k) => obj[k], parsed);
 
         var is_node = true;
-        if(typeof next_val === 'object' && next_val !== null && !Array.isArray(next_val)) {
+        if(typeof next_val === 'object' && next_val !== null) {
             is_node = false;
-            var keys = Object.keys(next_val);
 
-            if(known_nodes.indexOf(next_keypath[next_keypath.length - 1]) > -1){
-                is_node = true;
-            }
-
-            var keys_indicating_node = keys.filter(
-              e => indicates_node.indexOf(e) > -1
-            )
-
-            if(keys_indicating_node.length > 0){
-                is_node = true;
-            }
-
-            if(!is_node){
-                Object.keys(next_val).map(k =>
-                    remaining_paths.push(next_keypath.concat([k]))
-                );
+            if (Array.isArray(next_val)) {
+              next_val.forEach(function (item, index) {
+                remaining_paths.push(next_keypath.concat([index]))
+              });
+            } else {
```

Comment:
We can move `is_node` to the else statement

Reply:
Moving it into the else statement would skip dictionaries that have non-object members since in those cases the `next_val` type would not be `object`.